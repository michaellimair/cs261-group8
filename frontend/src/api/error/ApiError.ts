class ApiError<T> extends Error {
  data?: T;

  static defaultMessage = 'An error has occurred in the API.';

  constructor(message?: string, data?: T) {
    super(message ?? ApiError.defaultMessage);
    this.data = data;
  }

  static isApiError(obj: any): obj is ApiError<any> {
    return obj instanceof ApiError;
  }
}

export default ApiError;
