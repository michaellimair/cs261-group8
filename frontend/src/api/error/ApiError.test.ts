import ApiError from './ApiError';

describe('ApiError', () => {
  it('is instantiated successfully', () => {
    interface ISampleError {
      status_code: number;
      message: string;
    }

    const errorData = {
      status_code: 500,
      message: 'An error has occurred',
    };

    const err = new ApiError<ISampleError>(errorData.message, errorData);

    expect(err.data).toMatchObject(errorData);
    expect(err).toMatchSnapshot();
  });

  describe('isApiError', () => {
    it('identifies ApiError classes correctly', () => {
      const err = new ApiError<undefined>('Unknown error');

      expect(ApiError.isApiError(err)).toBe(true);
    });
  });
});
