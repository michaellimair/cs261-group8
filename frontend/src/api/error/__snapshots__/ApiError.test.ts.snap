// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiError is instantiated successfully 1`] = `[Error: An error has occurred]`;
